arbor/backends/gpu/reduce_by_key.hpp

```diff
@@ -11,80 +11,30 @@ namespace impl{
 
 constexpr unsigned mask_all = 0xFFFFFFFF;
 
-// Wrappers around the CUDA warp intrinsics used in this file.
-// CUDA 9 replaced the warp intrinsics with _sync variants, and
-// depricated the old symbols.
-// These wrappers can be removed CUDA 9 becomnse the minimum
-// version used by Arbor.
-
-template <typename T>
-static __device__ __inline__
-T arb_shfl(T x, unsigned lane) {
-#if __CUDACC_VER_MAJOR__ < 9
-    return __shfl(x, lane);
-#else
-    return __shfl_sync(mask_all, x, lane);
-#endif
-}
-
-template <typename T>
-static __device__ __inline__
-unsigned arb_shfl_up(T x, unsigned delta) {
-#if __CUDACC_VER_MAJOR__ < 9
-    return __shfl_up(x, delta);
-#else
-    return __shfl_up_sync(mask_all, x, delta);
-#endif
-}
-
-static __device__ __inline__
-unsigned arb_ballot(int pred) {
-#if __CUDACC_VER_MAJOR__ < 9
-    return __ballot(pred);
-#else
-    return __ballot_sync(mask_all, pred);
-#endif
-}
-
-// return the power of 2 that is less than or equal to i
+// Return the largest power of 2 that is less than i
+// Caveat: i must be non-zero.
```

Comment:
Just regarding the comment: this returns the largest power of 2 no greater than i, right?

Reply:
Yes.
Due to other simplifications, this is no longer needed, and has been removed.